data_set_distances_test.go

```diff
@@ -0,0 +1,35 @@
+package stats
+
+import (
+	"testing"
+)
+
+func TestDataSetDistances(t *testing.T) {
+	var err error
+	dataPointX := []float64{2, 3, 4, 5, 6, 7, 8}
```

Comment:
Since I want to check the output as well it might be easier if you adopt the table driven test pattern:

https://github.com/golang/go/wiki/TableDrivenTests